### PR TITLE
Handle Collections

### LINKED ISSUE #98: Handle (top level) collection of objects
if `params =  { donations: [{ …donation attributes here }] }` I'd like t be able to call something like `DonationDecanter.decant(params[:donations])` and have it decant the collection of individual donation_params

### PR DESCRIPTION
**Pull request by inveterateliterate**
## Items Addressed
Resolves #98 

## Author Checklist
- [x] Add unit test(s)
- [x] Update documentation (if necessary)
- [x] Update version in `version.rb` following [versioning guidelines](https://github.com/LaunchPadLab/opex-public/blob/master/gists/gem-guidelines.md#pull-requests-and-deployments)

cc @dpikt

### CODE REVIEW

File: spec/decanter/decanter_extensions_spec.rb [hunk 1]
```diff
@@ -3,13 +3,66 @@
...
+    let!(:args) { [{ foo: 'bar' }, { foo: 'baz' }] }
+    let(:options) { { } }
+    let(:dummy_class) { Class.new { include Decanter::Extensions } }
+
+    before(:each) do
+      allow(dummy_class).to receive(:decant_args).and_return(true)
+    end
+
+    it 'calls decant_args on each element in the collection' do
+      # expect(dummy_class).to receive(:decant_args).twice
```
Comment (inveterateliterate) [inline spec/decanter/decanter_extensions_spec.rb:50]: @chawes13  can either have this commented line and the next line be the spec, or leave what's uncommented. Which do you prefer as a representation of this spec?

File: README.md [hunk 2]
```diff
@@ -70,6 +71,37 @@ rails g decanter Trip name:string start_date:date end_date:date
...
+  end
+```
+
+**Control Over Decanting Collections**
+
+You can use the `is_collection` option for explicit control over decanting collections.
+
+`decanted_trip_params = TripDecanter.decant(trip_params[:trips], is_collection: true)`
+
+If this option is not provided, autedetect logic is used to determine if the providing incoming params holds a single object or collection of objects.
```
Comment (chawes13) [inline README.md:99]: Typo: `autodetect`

File: README.md [hunk 3]
```diff
@@ -70,6 +71,37 @@ rails g decanter Trip name:string start_date:date end_date:date
...
+        { name: 'Disney World', start_date: '12/24/2018', end_date: '12/28/2018' },
+        { name: 'Yosemeite', start_date: '5/1/2017', end_date: '5/4/2017' }
+      ]
+    }
+    decanted_trip_params = TripDecanter.decant(trip_params[:trips])
+    Trip.create(decanted_trip_params) # bulk create trips with decanted params
+  end
+```
+
+**Control Over Decanting Collections**
```
Comment (chawes13) [inline README.md:93]: Should this be an h4 (i.e., `####`)?

File: lib/decanter/extensions.rb [hunk 4]
```diff
@@ -36,12 +35,28 @@ def decant_create!(args, **options)
...
         if specified_decanter = options[:decanter]
           Decanter.decanter_from(specified_decanter)
         else
           Decanter.decanter_for(self)
         end.decant(args)
       end
+
+      private
+
+      ## leveraging the approach used in the [fast JSON API gem](https://github.com/Netflix/fast_jsonapi#collection-serialization)
```
Comment (chawes13) [inline lib/decanter/extensions.rb:55]: Is two hashes to start a comment indicative of something?

File: lib/generators/rails/decanter_generator.rb [hunk 5]
```diff
@@ -2,12 +2,12 @@ module Rails
...
       source_root File.expand_path('../templates', __FILE__)
-      check_class_collision :suffix => 'Decanter'
+      check_class_collision suffix: 'Decanter'
       ASSOCIATION_TYPES = [:has_many, :has_one, :belongs_to]
 
-      argument :attributes, :type => :array, :default => [], :banner => 'field:type field:type'
+      argument :attributes, type: :array, default: [], banner: 'field:type field:type'
 
-      class_option :parent, :type => :string, :desc => 'The parent class for the generated decanter'
+      class_option :parent, type: :string, desc: 'The parent class for the generated decanter'
```
Comment (chawes13) [inline lib/generators/rails/decanter_generator.rb:10]: Are these changes related to this new functionality?

File: spec/decanter/decanter_extensions_spec.rb (hunk 1 above)
Comment (chawes13) [reply spec/decanter/decanter_extensions_spec.rb:50]: I prefer what's left uncommented. You can remove this line!

**Review by chawes13 — COMMENTED** [review]
Code looks great. Few comments and questions on the docs/specs, but nothing major!

**Comment by chawes13** [comment]
@inveterateliterate Since #100 was merged, can you bump the version to `3.4.0`?

File: README.md (hunk 3 above)
Comment (inveterateliterate) [reply README.md]: it can be!

File: lib/decanter/extensions.rb (hunk 4 above)
Comment (inveterateliterate) [reply lib/decanter/extensions.rb]: nope I think it was a js feels

File: lib/generators/rails/decanter_generator.rb (hunk 5 above)
Comment (inveterateliterate) [reply lib/generators/rails/decanter_generator.rb:10]: nope - was just flipping through files to get my bearing and it stuck out to me

**Review by inveterateliterate — COMMENTED** [review]
Thanks @chawes13 updated per your comments!

**Review by chawes13 — APPROVED** [review]
🎉
